dashboard/app/models/script.rb

```diff
@@ -1229,7 +1229,7 @@ def summarize(include_lessons = true, user = nil, include_bonus_levels = false)
       has_verified_resources: has_verified_resources?,
       has_lesson_plan: has_lesson_plan?,
       curriculum_path: curriculum_path,
-      script_announcements: script_announcements,
```

Comment:
@davidsbailey do I need to keep a line that says `script_announcements: announcements` here. I know we have done that in the past but I'm not entirely clear on when its needed

Reply:
Hey Dani, good question. the best practice is to make sure we avoid undesirable behavior caused by old client code hitting new server code. I think a good rule for us to have as a team is, by default, for launched features, leave the old server code around for a deploy or two to give the client code time to catch up. this way, we can have a simple path forward without having to re-evaluate every time.

If you want to skip this step, there are a number of situations in which it is safe to do so. I will add another comment going into that, but for now I would just keep `script_announcements` around for a day or two.

Reply:
to recap, the problem with changing server and client code in the same PR is when the old client code is incompatible with the new server code. old client code can hit new server code in the following situations:
* a web page is opened before the deploy, the page stays open, then it makes an AJAX request after the deploy. or
* a server page has cache control headers which allow it to be cached. after the deploy, the browser continues receiving server responses with cached page, which contains old client code. when those pages load, they may make AJAX requests which hit new server code.
* during the deploy, new frontends have been started but old ones have not been shut down yet. a web request hits the old server (getting old JS), and the old JS then immediately makes an AJAX request, which happens to get routed to a new server.

if we want to take the time and can convince ourselves that we are in one of the following scenarios, then we could consider just removing the server code immediately:

1. user impact would be tolerably low, and no errors are generated. more specific considerations:

    1. unlaunched feature --> no user impact

    1. if the feature only exists on levelbuilder, and no problem persists after they hard-refresh the page (e.g. no data loss), we can call this low user impact

    1. the unexpected behavior is not too bad (e.g. missing script announcements in this case)

    1. "no errors generated" is important because it could lead to different behavior on different browsers, and also because it could clog up our monitoring.

1. the data is only sent down to the client as part of a server response (not part of an ajax request). In this case, all the data is in one server response, so there is no way for it to becoming from different places. 

If the data in question is part of a response to an AJAX request, it is theoretically possible that we could convince ourselves that there would never be a timing problem, but it seems hard to rule out the three problematic scenarios listed earlier so it is probably generally not worth going through the trouble of this.

Clear as mud?

Reply:
Thank you so much Dave for writing all of this up!